src/manifolds/ProductManifold.jl

```diff
@@ -153,6 +153,12 @@ function check_point(M::ProductManifold, p::Union{ProductRepr,ProductArray}; kwa
     (length(errors) == 1) && return cerr[1]
     return nothing
 end
+function check_point(M::ProductManifold, p; kwargs...)
+    return DomainError(
+        typeof(p),
+        "The point $p is not a point on $M, since currently only ProductRepr and ProductArray are suppored typoes for points on product manifolds",
```

Comment:
```suggestion
        "The point $p is not a point on $M, since currently only ProductRepr and ProductArray are supported types for points on product manifolds",
```